### ISSUE TITLE
Track download by getting the waypoints

### ISSUE THREAD

**Issue by aalemann** [danstoner/dailymile_export#14]
Hi, I really like your script. What I would like even more, would be an option to download the tracks you have recorded.

I looked at the page source of one of my entries (simply using Firefox --> 'View Page Source') and the tracked waypoints can be found ("var waypoints") there (same for the elevation profile, look for 'elevation_profile'). 

This would be awesome :)

**Comment by danstoner** [comment]
Thanks. I never use tracks so I hadn't looked at that at all. Looks pretty straightforward as the GPS tracks are available via the dailymile API.

http://www.dailymile.com/api/documentation#gps

**Comment by danstoner** [comment]
So what is the preferred output format for the track data?  Individual files that each contain one track?  Shoving the data into a column in a spreadsheet? How will the track data be used after downloading?

Thanks!

**Comment by aalemann** [comment]
wow, fast feedback :)

hm, individual files are nicer, they should be something simple like:

latitude,longitude,elevation,name
45.40252,-121.77589,1399,wp1
45.39254,-121.69946,2394,wp2 

You get latitude and longitude from the var waypoints, for each waypoint, it has the format [latitude,longitude] and the waypoints are seperated by a "," 
The elevation is stored in "elevation_profile", one value for each waypoint. The name can just be an increasing number.

Another option would be to save trackpoints:

trackpoint,latitude,longitude,alt,name
1,45.9874167,-76.8752333,79.8,Track_name
2,45.9860000,-76.8737833,111.4

As track name, one could use the id or the timestamp of the whole entry (which is already included in your csv-export-script). Unfortunately, dailymile seems not to save the time stamp for each trackpoint (this would allow to calculate speed) - at least, I have not fould it. 

In my opinion, trackpoints are preferable. (some more information on simple GPS files, can be found here: http://www.gpsvisualizer.com/tutorials/tracks.html)

Btw: I already thought that there is no possibility to get my 100s of runs (their tracks) back from dailymile, but now... there is hope again :) And I promise you to buy a drink if I will ever come to your city if this works (I don't see a reason why it should not work)

**Comment by danstoner** [comment]
This is a sample entry that includes a "map":

http://www.dailymile.com/people/danstoner/entries/35542790

The route was added via the map tool, not sure if this is the same data as would appear via a true GPS track.

**Comment by danstoner** [comment]
The route itself must be marked public for "Everyone" access.

Given a sample entry that includes a GPX track (note this is the www service not the api service):

http://www.dailymile.com/people/danstoner/entries/38403528

There will be a div named "routes_mini" that contains a link to the route identifier:

```
  <div id="routes_mini">
    <a href="/routes/1965264-running-route"><img alt="Staticmap?size=150x150&amp;sensor=false&amp;path=weight:2|color:0x0000ff|enc:cbftdhhhuny%40a%40%5dyw%5dmk%40e%5bsg%40i%40m%40_%40_%40%7d%40k%40%7bayw%40%40g%40cug_%40cia%40e%40yaa%5b%5dwiai%40mcy%40%3fo%40ms%40ye%40e%40mgo%40ebuxgbo%40abk%40%7dakacaq%40i%40k%40e%40qe%40eybeu%40isaa%5dfyputo%5cg%5eax%40ntf%40dn%40cz%5bjc%3ffhjn%40p%40gx%3fz%40nx%40tfavtbtvavifa_%40%5eije%3fgekq%40mqi%5bm%40ki%5dmuey%40yia%5b_%40imkaibgngtaf%5ehb%40tdc%60a%5e%5c%5cb%40vxrh%5cdddlb%40dflfzagt%40%3f%5ejx%40lvaxzdvat%40bncrr%40nf%40bf%40pv%40%7c%40j%40h%40ptdrbp%40aj%40sp%40yd%40q%40n%40%7b%40n%40e%40h%40_%40%7e%40yd%40%3fhv%40gzddxcdaad%40%40v%40hxa%60%40bdaclabv%40iv%40j%60%40ct%40bv%3fzsfmec%40a%7da%3fina%40dyhmb%40b%5egrofm%40gaoiuikgak%40m%40qi%40y%40qqa%40qobm%40a%40sw%40i%40iae%40sa_%40%7baosgm%40yqaykcc%5dduf_bl%40wreh%40dhhnfp%40jnbph%40%40pdlh%60%40%5cxzj%40f%40pd%5cb%7e%40txrz%40%5edhp%40zncvfbrfl%40zz%40zfah%40" src="http://maps.google.com/maps/api/staticmap?size=150x150&amp;sensor=false&amp;path=weight:2|color:0x0000ff|enc:cbftDhhhuNy%40a%40%5DYW%5DMk%40E%5BSg%40i%40m%40_%40_%40%7D%40k%40%7BAYw%40%40g%40CUG_%40CiA%40e%40YaA%5B%5DWiAi%40MCy%40%3Fo%40Ms%40Ye%40e%40MGO%40EBUXGBO%40aBk%40%7DAkAcAq%40i%40k%40e%40Qe%40EyBEu%40IsAA%5DFYPUTO%5CG%5EAX%40NTf%40Dn%40CZ%5BjC%3FFHJN%40p%40GX%3Fz%40Nx%40TfAVtBTVAVIfA_%40%5EIJE%3FGEKq%40MQI%5Bm%40KI%5DMUEy%40YiA%5B_%40IMKAIBGNGtAF%5EHb%40TdC%60A%5E%5C%5Cb%40VXRH%5CDDDLb%40DFLFzAGt%40%3F%5EJx%40LvAXZDVAt%40BnCRr%40Nf%40Bf%40Pv%40%7C%40j%40h%40PTDRBp%40Aj%40Sp%40Yd%40q%40n%40%7B%40n%40e%40h%40_%40%7E%40Yd%40%3FHv%40GzDDXCdAAd%40%40v%40HXA%60%40BdAClABv%40Iv%40J%60%40Ct%40BV%3FZSFMEc%40A%7DA%3FINa%40DYHMb%40B%5EGROFM%40GAOIUIKgAk%40m%40Qi%40y%40QQa%40QoBm%40a%40Sw%40i%40iAe%40sA_%40%7BAOSGm%40YqAYkCC%5DDUF_Bl%40WREH%40DHHNFp%40JnBPh%40%40PDLH%60%40%5CXZj%40f%40PD%5CB%7E%40TXRz%40%5EDHp%40ZnCVFBRFl%40Zz%40ZfAh%40" /></a>
    <a href="/routes/1965264-running-route" class="zoom" style="left: 0; bottom: 0"><span class="magnify_icon">&nbsp;</span></a>
  </div>
```

We just need the route id:

```1965264-running-route```

And then we can use the "routes" endpoint and add .gpx to the end:

```http://api.dailymile.com/routes/1965264-running-route.gpx```

These really don't seem particularly useful compared to other fitness tracking implementations since it is just the physical track (no time info).

I am thinking about adding the feature to download them.

Naming scheme will possibly be something like 1965264-running-route.38403528.gpx.

```
<route_id>.<entry_id>.gpx
```

**Comment by danstoner** [comment]
Feature added to master branch via commit 62407bbccbb3309b49f38d8c7f0d49e2eabc0291

This is not yet bundled into a Release as of the time of this post, but will be included in v1.4.
